ansible/playbooks/roles/grafana/tasks/password_change.yml

```diff
@@ -0,0 +1,40 @@
+---
+- name: Fail when grafana admin password isn't set
+  fail:
+    msg: "Please specify grafana admin password (grafana_security.admin_password)"
+  when:
+    - grafana_security.admin_password is undefined or grafana_security.admin_password | length == 0
+
+- name: Compare current grafana password set on the remote with the schema file
+  block:
+    - name: Set facts for fetched grafana.ini file
+      set_fact:
+        grafana_ini_path: /tmp/{{ inventory_hostname }}/etc/grafana/grafana.ini
+
+    - name: Fetch the grafana.ini file from the remote
+      fetch:
+        src: /etc/grafana/grafana.ini
+        dest: /tmp
+
+    - name: Change fetched grafana file ownership to root
+      delegate_to: localhost
+      file:
+        path: "{{ grafana_ini_path }}"
+        owner: root
+        group: root
+        mode: 'u+rx,g-rwx,o-rwx'
+
+    - name: Fetch the password
+      set_fact:
+        current_admin_password: "{{ lookup('ini', 'admin_password', section='security', file=grafana_ini_path) }}"
+
+    - name: Remove fetched grafana file
+      delegate_to: localhost
+      file:
+        path: "{{ grafana_ini_path }}"
+        state: absent
+
+    # Grafana admin password change is only available through the grafana-cli
+    - name: Change admin assword using grafana-cli
+      command: grafana-cli admin reset-admin-password "{{ grafana_security.admin_password }}"
+      when: grafana_security.admin_password != current_admin_password
```

Comment:
I have a feeling that only this task is enough without checks before. Anyways, if such checks are required and the file contains any secret, `slurp` module is a better choice.